pkg/terra/contract_reader.go

```diff
@@ -80,19 +82,19 @@ func parseAttributes(attrs []cosmosSDK.Attribute) (output types.ContractConfig,
 		case "latest_config_digest":
 			// parse byte array encoded as hex string
 			if err := HexToConfigDigest(value, &output.ConfigDigest); err != nil {
-				return types.ContractConfig{}, err
+				return types.ContractConfig{}, errors.Wrapf(err, "failed to parse attribute %q", key)
```

Comment:
nit: we could have a helper:

```suggestion
				return attrError(key)
```

Reply:
I actually had exactly that at first, but then the stack trace points to the helper. Can probably just add a proper error type to have it both ways.

Reply:
No worries, we can merge it as is :)

Reply:
np I've got some more changes incoming already